samples_project/Assets/SampleViewer/Samples/Measure/Measure.cs

```diff
@@ -0,0 +1,298 @@
+        currentUnit = UnitType.m;
+        unitTxt = " m";
+        UnitDropdown.onValueChanged.AddListener(delegate {
+            UnitChanged();
+        });
+        ClearButton.onClick.AddListener(delegate {
+            ClearLine();
+        });
+
+       
+    }
+    
+    void Update()
+    {
+ 
+        if (Input.GetKey(KeyCode.LeftShift) && Input.GetMouseButtonDown(0))
+        {
+            RaycastHit hit;
+            Ray ray = Camera.main.ScreenPointToRay(Input.mousePosition);
+
+            if (Physics.Raycast(ray, out hit))
+            {
+                var lineMarker = Instantiate(LineMarker, hit.point, Quaternion.identity, arcGISMapComponent.transform);
+
+                var geoPosition = HitToGeoPosition(hit);
+
+                lineMarker.GetComponent<ArcGISLocationComponent>().enabled = true;
+                lineMarker.GetComponent<ArcGISLocationComponent>().Position = geoPosition;
+                lineMarker.GetComponent<ArcGISLocationComponent>().Rotation = new ArcGISRotation(0, 90, 0);
+
+                var thisPoint = new ArcGISPoint(geoPosition.X, geoPosition.Y, geoPosition.Z, spatialRef);
+
+                
+                if (stops.Count > 0)
+                {
+                    lastStop = stops.Peek();
+                    lastStopLocation = lastStop.GetComponent<ArcGISLocationComponent>();
+                    lastPoint = new ArcGISPoint(lastStopLocation.Position.X, lastStopLocation.Position.Y, lastStopLocation.Position.Z, spatialRef);
+                    
+                    //calculate distance from last point to this point
+                    geodedicDistance += ArcGISGeometryEngine.DistanceGeodetic(lastPoint, thisPoint, new ArcGISLinearUnit(unit), new ArcGISAngularUnit(unitDegree), ArcGISGeodeticCurveType.Geodesic).Distance;
+                    GeodedicDistanceText.text = "Distance: "+ Math.Round(geodedicDistance, 3).ToString()+unitTxt;
+
+                    featurePoints.Add(lastStop);
+                    //interpolate middle points between last point and this point
+                    Interpolate(lastStop, lineMarker, featurePoints);
+                    featurePoints.Add(lineMarker);
+                   
+                   
+                }
+                //add this point to stops and also to feature points where stop is user-drawed, and feature points is a collection of user-drawed and interpolated
+                stops.Push(lineMarker);
+               
+            }
+        }
+        RenderLine(ref featurePoints);
+        RebaseLine();
+
+    }
+    
+    private void Interpolate(GameObject start, GameObject end, List<GameObject> featurePoints)
+    {
+        ArcGISLocationComponent startLocation = start.GetComponent<ArcGISLocationComponent>();
+        ArcGISLocationComponent endLocation = end.GetComponent<ArcGISLocationComponent>();
+
+        ArcGISPoint startPoint = new ArcGISPoint(startLocation.Position.X, startLocation.Position.Y, startLocation.Position.Z, spatialRef);
```

Comment:
You don't need to create a new Point every time. Just get the position from the location component.